lib/gadget/gadget.vala

```diff
@@ -304,6 +304,22 @@ namespace Frida.Gadget {
 				range: range
 			);
 		}
+
+#if ANDROID
+		construct {
+			if (executable_name.index_of ("app_process") >= 0) {
```

Comment:
Would be good to make this check a bit stricter, to avoid false positives:

```suggestion
			if (executable_name.has_prefix ("app_process")) {
```